.github/workflows/header-gen.yml

```
@@ -18,15 +29,30 @@ jobs:
       uses: actions/setup-node@v4
       with:
         node-version: '20'
+        registry-url: "https://npm.pkg.github.com"
+        cache: npm
+        cache-dependency-path: header-gen/package-lock.json
 
     - name: Install Dependencies
       run: npm ci
       working-directory: ./header-gen
 
+    - name: Update version
+      if: github.event_name != 'pull_request'
+      uses: microbit-foundation/npm-package-versioner-action@v1
+      with:
+        working-directory: ./header-gen
+
     - name: Build
       run: npm run build
       working-directory: ./header-gen
 
     - name: Run Jest Tests
       run: npm test
       working-directory: ./header-gen
+
+    - name: Publish
+      run: npm publish
+      if: github.ref == 'refs/heads/main' || github.event_name == 'release' && github.event.action == 'created'
```

The `github.ref == 'refs/heads/main'` part will also publish the package if a commit with changes to this CI script (and no changes to the header-gen code) is pushed to the main branch, is that right?

---

Shouldn't the path field prevent this action from running if no changes are made to ./header-gen/* ?

---

The path also includes changes to this CI script with `"**/header-gen.yml"`, no?

---

Yeah, OK. I think that makes sense though. If we tweak the build in the future, I'd expect it to run to test the output.

---

I guess the simplest thing here is just to publish and GH release
```suggestion
      if: github.event_name == 'release' && github.event.action == 'created'
```

---

I'm happy with simple.